v2/cf-nodejs-logging-support/src/lib/config/jwt-service.ts

```diff
@@ -0,0 +1,51 @@
+import Logger from "../logger/logger";
+
+const jwt = require("jsonwebtoken");
+const ENV_DYN_LOG_HEADER = "DYN_LOG_HEADER";
+const ENV_DYN_LOG_KEY = "DYN_LOG_LEVEL_KEY";
+const DEFAULT_DYN_LOG_LEVEL_HEADER = "SAP-LOG-LEVEL";
+
+export class JWTService {
+    // Read dyn. log level header name from environment var
+    private static headerName = process.env[ENV_DYN_LOG_HEADER];
+    private static dynLogLevelHeader = JWTService.headerName ? JWTService.headerName : DEFAULT_DYN_LOG_LEVEL_HEADER;;
+
+    constructor() {
+
+    }
+
+    static getDynLogLevelHeaderName() {
+        return this.dynLogLevelHeader;
+    }
+
+    setDynLogLevelHeader() {
+        JWTService.dynLogLevelHeader = JWTService.headerName ? JWTService.headerName : DEFAULT_DYN_LOG_LEVEL_HEADER;
+    }
+
+    // Binds the Loglevel extracted from JWT token to the given request logger
+    static bindDynLogLevel(token: any, logger: Logger): Logger {
+        // Read dyn log level key from environment var.
+        const dynLogLevelKey = process.env[ENV_DYN_LOG_KEY];
+        var payload = this.verifyAndDecodeJWT(token, dynLogLevelKey);
+
+        if (payload) {
+            logger.setLoggingLevel(payload.level);
```

Comment:
Instead of passing `logger` this function could also be modified to return the logging level. This would also avoid importing `Logger`.